src/lib/hooks/useScrollPositionHook.js

```diff
@@ -0,0 +1,31 @@
+import { useLayoutEffect } from 'react';
+import { getElementsPositionDifference } from '../services/elementPositionService';
+
+export const useScrollPosition = (effect, dependencies, contentEl, viewportEl, wait) => {
+  let throttleTimeout = null;
+
+  const callBack = () => {
+    effect(getElementsPositionDifference(contentEl, viewportEl));
+    throttleTimeout = null;
+  };
+
+  useLayoutEffect(() => {
+    const handleScroll = () => {
+      if (wait) {
+        if (throttleTimeout === null) {
+          // eslint-disable-next-line react-hooks/exhaustive-deps
+          throttleTimeout = setTimeout(callBack, wait);
+        }
+      } else {
+        callBack();
+      }
+    };
+
+    viewportEl.current.addEventListener('scroll', handleScroll);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    return () => viewportEl.current.removeEventListener('scroll', handleScroll);
```

Comment:
`clearTimeout ` need to be added here in case we unmount component earlier than setTimeout callback is called

Reply:
Fixed, see above.